openfisca_core/simulation_builder.py

```diff
@@ -0,0 +1,360 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals, print_function, division, absolute_import
+
+from datetime import date
+from collections import OrderedDict  # Only for Python 2
+
+import dpath
+import numpy as np
+
+from openfisca_core.commons import basestring_type
+from openfisca_core.errors import VariableNotFound, SituationParsingError, PeriodMismatchError
+from openfisca_core.indexed_enums import Enum
+from openfisca_core.periods import key_period_size, period as make_period
+from openfisca_core.simulations import Simulation
+from openfisca_core.tools import eval_expression
+
+
+class SimulationBuilder(object):
+
+    def build_from_dict(self, tax_benefit_system, input_dict, default_period = None, **kwargs):
```

Comment:
@Morendil @pblayo if `SimulationBuilder ` doesn't have a state anymore, is there any point to keep the `self` here? Shouldn't we just make these methods static with `@staticmethod`?

Reply:
It will have a state, eventually. Builders generally do; their state tends to be the elements of partially constructed objects prior to assembly, and ancillary things. It already looks like the "ancillary things" here will include a mapping between singular and plural names of entities (to be named, perhaps EntityDictionary).